### PR TITLE
[IMP] account: refactor account types

### PR DESCRIPTION
**Pull request by aliyatastemirova**
Task: 2856281

- Remove user_type_id, account.account.type model, internal_type
- Add account_type that is a simple selection field
- Move internal_group and include_initial_balance to account.account
- Because of these changes, type_control_ids on account.journal is also removed

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_payment_method.py [hunk 1]
```diff
@@ -101,9 +101,8 @@ class AccountPaymentMethodLine(models.Model):
         ondelete='restrict',
         domain=lambda self: "[('deprecated', '=', False), "
                             "('company_id', '=', company_id), "
-                            "('user_type_id.type', 'not in', ('receivable', 'payable')), "
-                            "'|', ('user_type_id', '=', %s), ('id', '=', parent.default_account_id)]"
-                            % self.env.ref('account.data_account_type_current_assets').id
+                            "('account_type', 'not in', ('asset_receivable', 'liability_paypable')), "
+                            "'|', ('account_type', '=', 'asset_current'), ('id', '=', parent.default_account_id)]"
```
Comment (smetl) [inline addons/account/models/account_payment_method.py:105]: the lambda is no longer needed here then

File: addons/account/models/account_journal.py [hunk 2]
```diff
@@ -77,25 +77,23 @@ def _default_invoice_reference_model(self):
...
     suspense_account_id = fields.Many2one(
         comodel_name='account.account', check_company=True, ondelete='restrict', readonly=False, store=True,
         compute='_compute_suspense_account_id',
         help="Bank statements transactions will be posted on the suspense account until the final reconciliation "
              "allowing finding the right account.", string='Suspense Account',
         domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), \
-                             ('user_type_id.type', 'not in', ('receivable', 'payable')), \
-                             ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_current_assets').id)
+                             ('account_type', 'not in', ('asset_receivable', 'liability_paypable')), \
+                             ('account_type', '=', 'asset_current')]")
```
Comment (smetl) [inline addons/account/models/account_journal.py:96]: the lambda is no longer needed here then

File: addons/account/models/account_journal.py [hunk 3]
```diff
@@ -150,16 +148,15 @@ def _default_invoice_reference_model(self):
         help="Used to register a profit when the ending balance of a cash register differs from what the system computes",
         string='Profit Account',
         domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), \
-                             ('user_type_id.type', 'not in', ('receivable', 'payable')), \
-                             ('user_type_id', 'in', %s)]" % [self.env.ref('account.data_account_type_revenue').id,
-                                                             self.env.ref('account.data_account_type_other_income').id])
+                             ('account_type', 'not in', ('asset_receivable', 'liability_paypable')), \
+                             ('account_type', 'in', ('income', 'income_other'))]")
```
Comment (smetl) [inline addons/account/models/account_journal.py:152]: the lambda is no longer needed here then

File: addons/account/models/account_journal.py [hunk 4]
```diff
@@ -150,16 +148,15 @@ def _default_invoice_reference_model(self):
...
     loss_account_id = fields.Many2one(
         comodel_name='account.account', check_company=True,
         help="Used to register a loss when the ending balance of a cash register differs from what the system computes",
         string='Loss Account',
         domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), \
-                             ('user_type_id.type', 'not in', ('receivable', 'payable')), \
-                             ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_expenses').id)
+                             ('account_type', 'not in', ('asset_receivable', 'liability_paypable')), \
+                             ('account_type', '=', 'expense')]")
```
Comment (smetl) [inline addons/account/models/account_journal.py:159]: the lambda is no longer needed here then

File: addons/account/models/company.py [hunk 5]
```diff
@@ -57,7 +57,7 @@ class ResCompany(models.Model):
         tracking=True,
         help="No users can edit journal entries related to a tax prior and inclusive of this date.")
     transfer_account_id = fields.Many2one('account.account',
-        domain=lambda self: [('reconcile', '=', True), ('user_type_id', '=', self.env.ref('account.data_account_type_current_assets').id), ('deprecated', '=', False)], string="Inter-Banks Transfer Account", help="Intermediary account used when moving money from a liquidity account to another")
+        domain=lambda self: [('reconcile', '=', True), ('account_type', '=', 'asset_current'), ('deprecated', '=', False)], string="Inter-Banks Transfer Account", help="Intermediary account used when moving money from a liqity account to another")
```
Comment (smetl) [inline addons/account/models/company.py:60]: the lambda is no longer needed here then

File: addons/account/models/company.py [hunk 6]
```diff
@@ -78,14 +78,13 @@ class ResCompany(models.Model):
     income_currency_exchange_account_id = fields.Many2one(
         comodel_name='account.account',
         string="Gain Exchange Rate Account",
-        domain=lambda self: "[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', id), \
-                             ('user_type_id', 'in', %s)]" % [self.env.ref('account.data_account_type_revenue').id,
-                                                             self.env.ref('account.data_account_type_other_income').id])
+        domain=lambda self: "[('account_type', 'not in', ('asset_receivable','liability_paypable','asset_cash', 'liability_credit_card')), ('deprecated', '=', False), ('company_id', '=', id), \
+                             ('account_type', 'in', ('income', 'income_other'))]")
```
Comment (smetl) [inline addons/account/models/company.py:82]: the lambda is no longer needed here then

File: addons/account/models/company.py [hunk 7]
```diff
@@ -78,14 +78,13 @@ class ResCompany(models.Model):
...
     expense_currency_exchange_account_id = fields.Many2one(
         comodel_name='account.account',
         string="Loss Exchange Rate Account",
-        domain=lambda self: "[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', id), \
-                             ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_expenses').id)
+        domain=lambda self: "[('account_type', 'not in', ('asset_receivable','liability_paypable','asset_cash', 'liability_credit_card')), ('deprecated', '=', False), ('company_id', '=', id), \
+                             ('account_type', '=', 'expense')]")
```
Comment (smetl) [inline addons/account/models/company.py:87]: the lambda is no longer needed here then

File: addons/account/models/res_config_settings.py [hunk 8]
```diff
@@ -21,16 +21,15 @@ class ResConfigSettings(models.TransientModel):
         related="company_id.income_currency_exchange_account_id",
         string="Gain Account",
         readonly=False,
-        domain=lambda self: "[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', company_id),\
-                             ('user_type_id', 'in', %s)]" % [self.env.ref('account.data_account_type_revenue').id,
-                                                             self.env.ref('account.data_account_type_other_income').id])
+        domain=lambda self: "[('account_type', 'not in', ('asset_receivable','liability_paypable','asset_cash','liability_credit_card')), ('deprecated', '=', False), ('company_id', '=', company_id),\
+                             ('account_type', 'in', ('income', 'income_other'))]")
```
Comment (smetl) [inline addons/account/models/res_config_settings.py:25]: the lambda is no longer needed here then

File: addons/account/models/res_config_settings.py [hunk 9]
```diff
@@ -21,16 +21,15 @@ class ResConfigSettings(models.TransientModel):
...
     expense_currency_exchange_account_id = fields.Many2one(
         comodel_name="account.account",
         related="company_id.expense_currency_exchange_account_id",
         string="Loss Account",
         readonly=False,
-        domain=lambda self: "[('internal_type', '=', 'other'), ('deprecated', '=', False), ('company_id', '=', company_id),\
-                             ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_expenses').id)
+        domain=lambda self: "[('account_type', 'not in', ('asset_receivable','liability_paypable','asset_cash','liability_credit_card')), ('deprecated', '=', False), ('company_id', '=', company_id),\
+                             ('account_type', '=', 'expense')]")
```
Comment (smetl) [inline addons/account/models/res_config_settings.py:32]: the lambda is no longer needed here then

File: addons/account/models/res_config_settings.py [hunk 10]
```diff
@@ -43,7 +42,7 @@ class ResConfigSettings(models.TransientModel):
         string='Bank Suspense Account',
         readonly=False,
         related='company_id.account_journal_suspense_account_id',
-        domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), ('user_type_id.type', 'not in', ('receivable', 'payable')), ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_current_liabilities').id,
+        domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), ('account_type', 'not in', ('asset_receivable', 'liability_paypable')), ('account_type', '=', 'liability_current')]",
```
Comment (smetl) [inline addons/account/models/res_config_settings.py:45]: the lambda is no longer needed here then

File: addons/account/models/res_config_settings.py [hunk 11]
```diff
@@ -52,7 +51,7 @@ class ResConfigSettings(models.TransientModel):
         string='Outstanding Receipts Account',
         readonly=False,
         related='company_id.account_journal_payment_debit_account_id',
-        domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), ('user_type_id.type', 'not in', ('receivable', 'payable')), ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_current_assets').id,
+        domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), ('account_type', 'not in', ('asset_receivable', 'liability_paypable')), ('account_type', '=', 'asset_current')]",
```
Comment (smetl) [inline addons/account/models/res_config_settings.py:54]: the lambda is no longer needed here then

File: addons/account/models/res_config_settings.py [hunk 12]
```diff
@@ -62,13 +61,13 @@ class ResConfigSettings(models.TransientModel):
         string='Outstanding Payments Account',
         readonly=False,
         related='company_id.account_journal_payment_credit_account_id',
-        domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), ('user_type_id.type', 'not in', ('receivable', 'payable')), ('user_type_id', '=', %s)]" % self.env.ref('account.data_account_type_current_assets').id,
+        domain=lambda self: "[('deprecated', '=', False), ('company_id', '=', company_id), ('account_type', 'not in', ('asset_receivable', 'liability_paypable')), ('account_type', '=', 'asset_current')]",
```
Comment (smetl) [inline addons/account/models/res_config_settings.py:64]: the lambda is no longer needed here then

File: addons/account/models/res_config_settings.py [hunk 13]
```diff
@@ -62,13 +61,13 @@ class ResConfigSettings(models.TransientModel):
...
         help='Outgoing Payments are posted on an Outstanding Payments Account. '
              'In the bank reconciliation widget, they appear as blue lines.\n'
              'Bank transactions are then reconciled on the Outstanding Payments Account rather the Payable Account.')
     transfer_account_id = fields.Many2one('account.account', string="Internal Transfer Account",
         related='company_id.transfer_account_id', readonly=False,
-        domain=lambda self: [('reconcile', '=', True), ('user_type_id', '=', self.env.ref('account.data_account_type_current_assets').id)],
+        domain=lambda self: [('reconcile', '=', True), ('account_type', '=', 'asset_current')],
```
Comment (smetl) [inline addons/account/models/res_config_settings.py:70]: the lambda is no longer needed here then

File: addons/account/models/account_account.py [hunk 14]
```diff
@@ -64,10 +35,45 @@ def _check_user_type_id_unique_current_year_earning(self):
...
-    internal_group = fields.Selection(related='user_type_id.internal_group', string="Internal Group", store=True, readonly=True)
+    account_type = fields.Selection(
+        selection=[
+            ("asset_receivable", "Receivable"),
+            ("asset_cash", "Bank and Cash"),
+            ("asset_current", "Current Assets"),
+            ("asset_non_current", "Non-current Assets"),
+            ("asset_prepayments", "Prepayments"),
+            ("asset_fixed", "Fixed Assets"),
+            ("liability_paypable", "Payable"),
```
[the remark this answers is not in the document]
Comment (aliyatastemirova) [reply addons/account/models/account_account.py:46]: oh my god :laughing:

File: addons/account/report/account_invoice_report.py [hunk 15]
```diff
@@ -86,6 +86,7 @@ def _select(self):
                 line.company_id,
                 line.company_currency_id,
                 line.partner_id AS commercial_partner_id,
+                account.account_type,
```
[the remark this answers is not in the document]
Comment (aliyatastemirova) [reply addons/account/report/account_invoice_report.py:89]: I wanted to compensate `LEFT JOIN account_account_type user_type ON user_type.id = account.user_type_id` on line 119, but I am not sure if this is being used anywhere

**Comment by smetl** [comment]
@aliyatastemirova Runbot is :boom:

**Comment by aliyatastemirova** [comment]
@smetl fixed it :slightly_smiling_face:

**Comment by smetl** [comment]
@robodoo r+
@aliyatastemirova Very nice job! :100:

File: addons/account/models/account_account.py [hunk 16]
```diff
@@ -366,6 +370,17 @@ def _compute_is_off_balance(self):
...
 
+    @api.depends('account_type')
+    def _compute_include_initial_balance(self):
+        for account in self:
+            account.include_initial_balance = account.account_type not in ('income', 'income_other', 'expense', 'expense_depreciation', 'expense_direct_cost', 'off_balance')
+
+    @api.depends('account_type')
+    def _compute_internal_group(self):
+        for account in self:
+            if account.account_type:
```
Comment (smetl) [inline addons/account/models/account_account.py:381]: you need to remove this line
@robodoo r-

**Comment by smetl** [comment]
@robodoo r+

File: addons/l10n_ma/data/l10n_ma_chart_data.xml [hunk 17]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="ISO-8859-15"?>
```
Comment (KangOl) [inline addons/l10n_ma/data/l10n_ma_chart_data.xml:1]: Encoding issue. Change the header to match the encoding of the file.

File: addons/account/views/account_account_views.xml [hunk 18]
```diff
@@ -59,13 +59,13 @@
                             <page name="accounting" string="Accounting">
                                 <group>
                                     <group>
-                                        <field name="user_type_id" widget="account_hierarchy_selection"/>
```
[the remark this answers is not in the document]
Comment (aliyatastemirova) [reply addons/account/views/account_account_views.xml:62]: I was thinking to update the widget to work with the new `account_type` after merging this task, would it be ok?

**Comment by csnauwaert** [comment]
@robodoo r+

**Comment by csnauwaert** [comment]
@robodoo r+

File: addons/account/views/account_account_views.xml (hunk 18 above)
[the remark this answers is not in the document]
Comment (aliyatastemirova) [reply addons/account/views/account_account_views.xml:62]: I have already started adapting it to `account_type`

**Pull request closed without merging**